### PR TITLE
add DynamicSparsityPattern::compute_Tmmult_pattern() by moving existi…

### PR DESCRIPTION
**Pull request by davydden**
…ng code

### CODE REVIEW

File: include/deal.II/lac/dynamic_sparsity_pattern.h [hunk 1]
```diff
@@ -446,6 +446,14 @@ class DynamicSparsityPattern : public Subscriptor
   compute_mmult_pattern(const SparsityPatternTypeLeft & left,
                         const SparsityPatternTypeRight &right);
 
+  /**
+   * Construct and store in this object the sparsity pattern corresponding to
+   * the product of transposed @p left and @p right sparsity pattern.
```
Comment (jppelteret) [inline include/deal.II/lac/dynamic_sparsity_pattern.h]: To ensure unambiguity, should we write this as `transposed @p left and non-transpose @p right`, or something to that effect?

**Comment by jppelteret** [comment]
Want to squash your commits so that we can merge this?

**Comment by davydden** [comment]
> Want to squash your commits so that we can merge this?

done.

**Pull request merged**